pkg/provisioner/provisioner.go

```diff
@@ -83,6 +83,10 @@ type Provisioner interface {
 	// PowerOff ensures the server is powered off independently of any image
 	// provisioning operation.
 	PowerOff() (result Result, err error)
+
+	// IsReady checks if the provisioning backend is available to accept
+	// all the incoming requests.
+	IsReady() (result Result, err error)
```

Comment:
I'm sceptical about returning a provisioner.Result here. It's even less obvious than usual what the 'Dirty' flag means. A function named something like `IsReady` should return a bool.
I realise this is done so that we can return a RequeueAfter value, but I'm not sure that we really need that to be customisable by the provisioner.

Reply:
I suggested the provisioner.Result in part to be consistent with the other methods, but I see your point.

Reply:
I reverted back the `IsReady()` signature